libcudacxx/include/cuda/std/__cccl/compiler.h

```diff
@@ -62,6 +68,7 @@
 
 // clang-cuda does not define __CUDACC_VER_MAJOR__ and friends
 #if defined(_CCCL_CUDA_COMPILER_CLANG)
+#  include "cuda.h" // CUDA_VERSION
```

Comment:
I am unsure about pulling in the CUDA driver header. I would appreciate a second opinion here.

Reply:
Definitely agree. Let me see if I can find an alternative.

Reply:
checked. Including the `cuda.h` header is the standard way to detect the version, see also https://github.com/microsoft/STL/issues/1949.
A bad (I guess) alternative could be
```cpp
#if defined(CUDA_VERSION)
#  define _CCCL_CUDACC_VER_MAJOR CUDA_VERSION / 1000
#else
#  define _CCCL_CUDACC_VER_MAJOR "Please include `cuda.h`" @@or some kind of error@@
#endif
```

Reply:
thanks for pointing me out this detail. Found a good reference https://github.com/llvm-mirror/clang/blob/master/lib/Headers/__clang_cuda_runtime_wrapper.h 
>  * WARNING: This header is intended to be directly -include'd by the compiler and is not supposed to be included by users.

`__clang_cuda_runtime_wrapper.h` already includes `cuda.h`. Thanks

Reply:
removed cuda.h and added a comment